### PR TITLE
feat(router): add option to wrap vk router with another implementing http.Handler

### LINKED ISSUE #77: Add in opentelemetry capabilities
Most of our services are using vektor for routing.

Opentelemetry's http instrumentation works my wrapping the mux of whatever we want to instrument (in our case, vektor) into otel's mux.

Vektor does not expose the mux for other services to wrap, which means we have to configure vektor in such a way that it becomes wrapped, optionally, and configured as per whichever other project is using it.

Sibling issue of https://github.com/suborbital/scn/issues/221

### PR DESCRIPTION
**Pull request by javorszky**
Closes #77

This is in preparation so we can use the functionality in atmo to decorate the router with the opentel muxer

### CODE REVIEW

File: vk/server.go
Comment (rnpridgeon) [inline vk/server.go:103]: Do we need to keep a reference to the original router and if so are we certain it never needs to be finalized?

File: vk/server.go
Comment (rnpridgeon) [inline vk/server.go:33]: should internalRouter also be quiet route aware

**Review by rnpridgeon — APPROVED** [review]
Seems reasonable, I did ask a few questions mostly for my own understanding but nothing I think was worth blocking the PR on. Probably a good idea to have @cohix give it a quick pass before merging though.

File: vk/server.go
Comment (javorszky) [reply vk/server.go:33]: the `router` here is the built in vektor one. This is the one that gets used as internalRouter. I'll change the name to make this less confusing

File: vk/server.go
Comment (javorszky) [reply vk/server.go:103]: the `Finalize` has a `sync.Once` in it, so even if we repeatedly call it, it will only ever run once. The decision for why this is so is with @cohix, and I don't know enough about the context of why it was built in such a way yet to answer this :)

File: vk/options.go
Comment (cohix) [inline vk/options.go:41]: Normally I'd do this in `opts.finalize`, but not a big deal either way

File: vk/server.go
Comment (cohix) [inline vk/server.go:36]: Don't think we need to set this here, since it gets set up in `Start` ?

File: vk/test/routes_test.go
Comment (cohix) [inline vk/test/routes_test.go:17]: Curious why a suite is useful here?

File: vk/server.go
Comment (javorszky) [reply vk/server.go:36]: Good spot, thank you! Removed

File: vk/options.go
Comment (javorszky) [reply vk/options.go:41]: The reason I did it here is because one of the options is about the router wrapper. If I declare the default here, all I have to do is loop through the passed in router wrapper, which will reassign it to what the user wanted, and we're done, saving us an extra if check in finalise.

Somewhat related, I think this should be cleaned up, because the configuration, or the changing of bits of the router / server happens in all of these places:
1. server.New
2. newOptsWithModifiers
3. server.Start
4. server.finalize

Keeping track of what gets set where is somewhat confusing. Ideally everything would be done in the New with their default values, then the option modifiers loop through, and then the only thing Start does is start listening on the configured places

File: vk/test/routes_test.go
Comment (cohix) [reply vk/test/routes_test.go:17]: Cool!

**Pull request merged**
